core/che-core-api-core/src/main/java/org/eclipse/che/api/core/util/LimitedSizeLineConsumerWrapper.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.core.util;
+
+import java.io.IOException;
+
+/**
+ * Wrapper of {@link LineConsumer} which limits total size of consuming lines.
+ *
+ * @author Dmytro Nochevnov
+ * @author Anatolii Bazko
+ */
+public class LimitedSizeLineConsumerWrapper implements LineConsumer {
+  private final int limit;
+  private final LineConsumer lineConsumer;
```

Comment:
Why you chouse wrapping instead of extending?

Reply:
Because @tolusha suggested wrapping [here](https://github.com/eclipse/che/pull/8652/files/1f18703c2ee9bcc393593ffc5861542425f58534#r167779494) and it actually looks better than mine as [you mentioned later](https://github.com/eclipse/che/pull/8652/files/1f18703c2ee9bcc393593ffc5861542425f58534#r167779494).

Reply:
Could you explain why did you add `Wrapper` suffix ?
https://en.wikipedia.org/wiki/Chain-of-responsibility_pattern

Reply:
Because it looks and behaves like a **structural** pattern _Wrapper_ kind of _Decorator_ which extends functionality of wrapping object of type **LineConsumer** dynamically - in our case is limiting total size of consuming lines and sending all messages to wrapping object being a type of  wrapping object.
Whereas _Chain of Responsibility_ pattern is **behavioral** pattern which chains the receiving objects and pass the request along the chain until an object handles it. So, it could extends interface of receivers of object only, not the interface of object to send.
More details about difference between _Decorator_ and _Chain of Responsibility_ patterns can be found here: https://coderanch.com/t/668194/certification/Difference-Decorator-Chain-Responsibility-patterns
An examples in Eclipse Che codebase:
- [IndentWrapperLineConsumer](https://github.com/eclipse/che/blob/2c2bd6637c8169018e2b33397f9d738bbf82a0b1/core/che-core-api-core/src/main/java/org/eclipse/che/api/core/util/IndentWrapperLineConsumer.java#L20)
- [WebSocketJsoWrapper.java](https://github.com/eclipse/che/blob/2c2bd6637c8169018e2b33397f9d738bbf82a0b1/ide/commons-gwt/src/main/java/org/eclipse/che/ide/websocket/impl/WebSocketJsoWrapper.java#L20)